src/apps/config/leader.lua

```diff
@@ -418,21 +425,23 @@ end
 
 function Leader:update_configuration (update_fn, verb, path, ...)
    self:notify_pre_update(self.current_configuration, verb, path, ...)
-   local to_restart =
-      self.support.compute_apps_to_restart_after_configuration_update (
-         self.schema_name, self.current_configuration, verb, path,
-         self.current_in_place_dependencies, ...)
-   local new_config = update_fn(self.current_configuration, ...)
-   local new_app_graph = self.setup_fn(new_config)
-   local actions = self.support.compute_config_actions(
-      self.current_app_graph, new_app_graph, to_restart, verb, path, ...)
-   self:enqueue_config_actions(actions)
-   self.current_app_graph = new_app_graph
-   self.current_configuration = new_config
-   self.current_in_place_dependencies =
-      self.support.update_mutable_objects_embedded_in_app_initargs (
-         self.current_in_place_dependencies, self.current_app_graph,
-         verb, path, ...)
+   local new_graph = update_fn(self.current_configuration, ...)
```

Comment:
Is this right?  It looks like before, update_fn returned a configuration, whereas now you are having it return a graph.  I looked through the rest of the file for relevant changes but I must have missed them.